File: src/main/java/br/com/blz/testjava/http/data/request/WarehouseRequest.java
package br.com.blz.testjava.http.data.request;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class WarehouseRequest {

    @NotNull
    private String locality; //Deveria ser um Enum?
    @NotNull(message = "quantity is required")
    @Min(value = 0, message = "quantity must be a positive number")
    private Integer quantity;
    @NotNull(message = "warehouse type is required")
    private WarehouseTypeRequest type;
}

File: src/main/kotlin/com/boticario/testkotlin/application/request/WarehouseRequest.kt
package com.boticario.testkotlin.application.request

import com.boticario.testkotlin.domain.entity.Warehouse
import com.boticario.testkotlin.domain.enum.WarehouseType

data class WarehouseRequest(
        val locality: String,
        val quantity: Int,
        val type: WarehouseType
) {
    fun toDomain() = Warehouse(locality, quantity, type)
}
